flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2711,7 +2717,9 @@ class ScalarExprLowering {
         continue;
       }
       const bool actualArgIsVariable = Fortran::evaluate::IsVariable(*expr);
-      if (arg.passBy == PassBy::BaseAddress || arg.passBy == PassBy::BoxChar) {
+      if (arg.passBy == PassBy::BaseAddress ||
+          arg.passBy == PassBy::BaseAddressValueAttribute ||
```

Comment:
I would not stick this case here since all the copy-in/copy-out part is unrelated to this case. I would just place it above. 
```
if (arg.passBy == PassBy::BaseAddressValueAttribute) {
} else if (arg.passBy == PassBy::BaseAddress || arg.passBy == PassBy::BaseAddressValueAttribute) {
} else if ...
```

Reply:
Yeah, makes sense. Moved it up from that portion.